### PR TITLE
update doc partially

### PR DESCRIPTION
**Pull request by henrique-lh**
## Description

With this pr the codex documentation will be updated

## Related issues

https://linear.app/datarisk/issue/NEODV-957/[techdebt]-atualizar-o-guia-de-uso-do-codex

### CODE REVIEW

**Review by johnrc90 — CHANGES_REQUESTED** [review]

File: docs/source/training_guide.rst [hunk 1]
```diff
@@ -45,38 +51,38 @@ For the custom experiment you need a entrypoint function like this:
...
         # if my_var is None:
-        #    raise Exception("Could not find `env` variable")
+        #   raise Exception("Could not find `env` variable")
 
-        ## Variável de ambiente carregada do Neomaril com nome da base de dados (usado em alternativa a linha 61)
+        ## Environment variable loaded from Neomaril with database name (used alternatively to line 61)
         # df = pd.read_csv(base_path+'/'+os.getenv('inputFileName'))
-        df = pd.read_csv(base_path+"/dados.csv")    # Carrega a base de dados (dados.csv) que precisa ter o mesmo nome
-                                                    # arquivo enviado para o Neomaril no campo 'train_data'
+        df = pd.read_csv(base_path+"/dados.csv")    # Load the database (data.csv) which must have the same name
```
Comment (ManuDiasCruz) [inline docs/source/training_guide.rst:67]: Here should be "dados.csv" or you should change the parameter name that goes inside the `read_csv(...)`
Also... 
"# Load the database (data.csv) which must have the same filename
"# sent to Neomaril in the 'train_data' field

File: docs/source/index.rst [hunk 2]
```diff
@@ -64,7 +64,7 @@ Maintainers
 
 - Manuella Dias Cruz: https://github.com/ManuDiasCruz
 - Jonathan Carvalho: https://github.com/johnrc90
-
+- Guilherme Cardoso: https://github.com/gkardoso
```
Comment (ManuDiasCruz) [inline docs/source/index.rst:67]: I think you can also adds your name here @henrique-lh

File: docs/source/connecting_datasource.rst [hunk 3]
```diff
@@ -0,0 +1,90 @@
+Data Source
+===========
+
+It is possible to connect a cloud provider to Neomaril. Currently, Neomaril supports the following providers:
```
Comment (ManuDiasCruz) [inline docs/source/connecting_datasource.rst:4]: I think it is important here to make clear that you will could use this cloud provider data sources as data entrance at training experiments. Also @gkardoso do we already have this adaptation to enter the dataset_hash here? I didn't saw at the code. We can create a issue to make it :smile:

File: docs/source/connecting_datasource.rst [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+It generates a DHash
+
+If you already connected your data source to Neomaril and imported a data set, you can get your data set using DHash:
+
+.. code:: python
+    dataset = datasource.get_dataset(dataset_hash='D66c8bc440dc4882bfeff40c0dac11641c3583f3aa274293b15ed5db21000b49')
+
+    # >>> 2024-03-20 19:20:16.688 | INFO     | neomaril_codex.base:__init__:20 - Loading .env
+    # >>> 2024-03-20 19:20:16.692 | INFO     | neomaril_codex.base:__init__:30 - Successfully connected to Neomaril
+
```
[the remark this answers is not in the document]
Comment (ManuDiasCruz) [reply docs/source/connecting_datasource.rst:77]: I agree!

**Review by ManuDiasCruz — CHANGES_REQUESTED** [review]
Great improvements! But we still can work on it. I agree with Carvalho, that we can add more how to use things at it. 
Also, I know it isn't the purpose of this issue, but we can create a issue to improve some inner code comments, completing validation rules for example.

File: docs/source/index.rst (hunk 2 above)
Comment (henrique-lh) [reply docs/source/index.rst:67]: I'm not sure, I just changed the docs haha

File: docs/source/connecting_datasource.rst (hunk 3 above)
Comment (henrique-lh) [reply docs/source/connecting_datasource.rst:4]: What do you think about:
"It is possible to connect a cloud provider to Neomaril. The cloud provider is a source of data, where you can store and get the data that is saved data to create/perform models.

Currently, Neomaril supports the following providers:..."?

Comment (henrique-lh) [reply docs/source/connecting_datasource.rst]: [Resolved](https://github.com/datarisk-io/mlops-neomaril-codex/pull/40/commits/8e091cec28681f5374f809e36a78d2e02a7d8929) (I think)

File: docs/source/training_guide.rst [hunk 5]
```diff
@@ -15,18 +15,24 @@ First we need to create a training experiment. We aggregate multiple train runs
 
 **AutoML:** Is when the training is pre defined using the AutoML module. The required files are the training data and some configuration parameters for the package.
 
+**External:** Is when training information of models that have been trained externally in your local machine.
```
[the remark this answers is not in the document]
Comment (henrique-lh) [reply docs/source/training_guide.rst]: [Resolved](https://github.com/datarisk-io/mlops-neomaril-codex/pull/40/commits/f39c9d25834639c6f216e275544bf5e4b679486f)

File: docs/source/connecting_datasource.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (henrique-lh) [reply docs/source/connecting_datasource.rst:79]: [Resolved](https://github.com/datarisk-io/mlops-neomaril-codex/pull/40/commits/7ee8e26add26f9803d2bd4f77ea7189ee9f44b6d)

File: docs/source/training_guide.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (henrique-lh) [reply docs/source/training_guide.rst]: [Resolved](https://github.com/datarisk-io/mlops-neomaril-codex/pull/40/commits/7ee8e26add26f9803d2bd4f77ea7189ee9f44b6d)

File: docs/source/connecting_datasource.rst (hunk 3 above)
Comment (ManuDiasCruz) [reply docs/source/connecting_datasource.rst]: Not exactly... I will talke to Gui and checki it :smile:

Comment (ManuDiasCruz) [reply docs/source/connecting_datasource.rst]: This is done!

**Review by ManuDiasCruz — APPROVED** [review]
Great job Luís!

**Review by johnrc90 — APPROVED** [review]

**Pull request merged**